bin/exec.ml

```diff
@@ -111,19 +111,34 @@ let build_prog ~no_rebuild ~prog p =
     p
 ;;
 
-let get_path_and_build_if_necessary sctx ~no_rebuild ~dir ~prog =
+let dir_of_context common context =
```

Comment:
Maybe call context `ctx_name` because it's a `Context_name.t` and what is later called `context` is `Context.t`. Reading the Dune code this keeps being confusing, so for reviewability sake I'd prefer being more descriptive with names.